iceoryx_posh/include/iceoryx_posh/popo/sample.hpp

```diff
@@ -27,6 +28,41 @@ namespace popo
 template <typename T>
 class PublisherInterface;
 
+namespace internal
+{
+/// @brief helper struct for sample
+template <typename T>
+struct SamplePrivateData
+{
+    SamplePrivateData(cxx::unique_ptr<T>&& samplePtr, PublisherInterface<T>& publisher);
+
+    SamplePrivateData(SamplePrivateData&& rhs);
+    SamplePrivateData& operator=(SamplePrivateData&& rhs);
+
+    SamplePrivateData(const SamplePrivateData&) = delete;
+    SamplePrivateData& operator=(const SamplePrivateData&) = delete;
+
+    cxx::unique_ptr<T> samplePtr{[](T* const) {}}; // Placeholder. This is overwritten on sample construction.
+    std::reference_wrapper<PublisherInterface<T>> publisherRef;
+};
+
+/// @brief specialization of helper struct for sample for const T
+template <typename T>
+struct SamplePrivateData<const T>
+{
+    SamplePrivateData(cxx::unique_ptr<const T>&& samplePtr) noexcept;
+
+    SamplePrivateData(SamplePrivateData&& rhs);
+    SamplePrivateData& operator=(SamplePrivateData&& rhs);
+
+    SamplePrivateData(const SamplePrivateData&) = delete;
+    SamplePrivateData& operator=(const SamplePrivateData&) = delete;
+
+    cxx::unique_ptr<const T> samplePtr{
```

Comment:
This is just weird. The `cxx::unique_ptr` requires a default ctor which should be used here.

Reply:
initialized the pointer in the ctor initialization list